src/Common/src/TypeSystem/Interop/IL/Marshaller.cs

```diff
@@ -406,7 +406,7 @@ protected override TypeDesc MarshalArgument(TypeDesc managedType, ILEmitter emit
                 TypeDesc resolvedType = ((ByRefType)managedType).ParameterType;
 
                 var nativeType = PInvokeMethodData.Context.GetWellKnownType(WellKnownType.IntPtr).MakeByRefType();
-                var vPinnedOutValue = emitter.NewLocal(nativeType, true);
+                var vPinnedOutValue = emitter.NewLocal(PInvokeMethodData.Context.GetWellKnownType(WellKnownType.IntPtr), true);
```

Comment:
It does not make to pin the IntPtr. It is allocated on the local stack, so it has a fixed address by design.